resolwe/storage/manager.py

```diff
@@ -46,23 +55,37 @@ def process_delete(self):
                     file_storage = self._lock_file_storage(file_storage.id)
                     if file_storage is None:
                         continue
-                    logger.debug(
-                        "Deleting data from location {}".format(connector_name)
-                    )
-                    try:
-                        self.delete_single_location(file_storage, connector_name)
-                    except Exception:
-                        logger.exception(
-                            "Error deleting data from StorageLocation instance",
-                        )
+                    # Some other storage manager might have already processed this storage
+                    # location before we did.
+                    if not file_storage.storage_locations.filter(
```

Comment:
This check is unnecessary; it only produces an extra query. It can fail only inside the `self.delete_single_location`, and it already does the same check.

Reply:
Actually, I was not sure where exactly to place this check. In the end, I forgot to remove the redundant one :/